### PR TITLE
Git WorkTree APIs

### PR DESCRIPTION
**Pull request by outofambit**
## Overview

supports #6960
based on #7822

## Description

- adds functions for creating, listing, and removing git work trees
- adds [`jest-extended`](https://github.com/jest-community/jest-extended) for better assertions around strings (paths)

### CODE REVIEW

File: app/src/lib/git/worktree.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  return workTree
+}
+
+/** Cleanup the temporary worktree at a given location */
+export async function removeWorkTree(
+  repository: Repository,
+  path: string
+): Promise<true> {
+  await git(
```
Comment (iAmWillShepherd) [inline app/src/lib/git/worktree.ts]: If the worktree is not able to be removed, is returning `true` still cool?

File: app/src/lib/git/worktree.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    // NodeJS doesn't seem to include this when we ask for the temporary
+    // directory for the OS
+    const directoryName = Path.basename(t.path)
+    return directoryName.startsWith(DesktopWorkTreePrefix)
+  })
+}
+
+/**
+ * Lookup or create a new temporary work tree for use in Desktop without
+ * touching the user's current repository state.
```
Comment (iAmWillShepherd) [inline app/src/lib/git/worktree.ts]: I think a note here that this function also updates `HEAD` due to a checkout would be nice.

File: app/src/lib/git/worktree.ts (hunk 1 above)
Comment (outofambit) [reply app/src/lib/git/worktree.ts]: if any errors are occurred dugite should throw an error, in which case the caller would not get `true` returned so i think this is ok. does that sound right?

File: app/src/lib/git/worktree.ts (hunk 2 above)
Comment (outofambit) [reply app/src/lib/git/worktree.ts]: done in 96a12fe3d146ef187689fa9eb891eedf018ec4b5

**Comment by iAmWillShepherd** [comment]
@outofambit feel free to merge when ready 👍

**Pull request merged**